### DISCUSSION TITLE
Extractor & Maps, VMaps, MMaps

### DISCUSSION THREAD

**Discussion by ZaDarkSide** [TrinityCore/TrinityCore#26868] [Suggestions]
Here's an ideea instead of re-extracting the damn data from the client each and every time when extractor/collision code changes or a slightly better algorithm/heuristic/whatever gets implemented that _might_ not produce like huge holes in the collision meshes or other hard to catch _bugs_, maybe it would be a better ideea that someone should generate them once, make appropriate fixes to the meshes by hand if necessary and release a _stable_ version like once a year/month/etc like TDB.

It's not like the geometry data gonna change in all maps all the time. For all intents and purposes the map data is static and it's just gets added to it over time when new expansions come out, also you only need at most to support 2 release versions for master and 3.3.5.

I get it, the extractor thingy is great to take the tedious work out of the way by automating most of the stuff, but it will never generate perfect nav meshes etc. this is my conclusion at least from my little understanding of the subject and seems to be like a recurring problem since forever and people are still tweaking the extractor/collision code even today.

If the ideea is dumb or not feasible please describe why. Thx.

**Comment by Aokromes** [comment]
> Here's an ideea instead of re-extracting the damn data from the client each and every time when extractor/collision code changes or a slightly better algorithm/heuristic/whatever gets implemented that _might_ not produce like huge holes in the collision meshes or other hard to catch _bugs_, maybe it would be a better ideea that someone should generate them once, make appropriate fixes to the meshes by hand if necessary and release a _stable_ version like once a year/month/etc like TDB.
> 
> It's not like the geometry data gonna change in all maps all the time. For all intents and purposes the map data is static and it's just gets added to it over time when new expansions come out, also you only need at most to support 2 release versions for master and 3.3.5.
> 
> I get it, the extractor thingy is great to take the tedious work out of the way by automating most of the stuff, but it will never generate perfect nav meshes etc. this is my conclusion at least from my little understanding of the subject and seems to be like a recurring problem since forever and people are still tweaking the extractor/collision code even today.
> 
> If the ideea is dumb or not feasible please describe why. Thx.

if we had a perfect way to fix everything we don't had to push fixes for it, also, even blizzard have maps/vmaps/mmaps bugs.

**Reply by ZaDarkSide** [reply]
Fixing meshes by hand is tedious but usually can fix any problems, depends what time you want to put into fixing them, that's why passionate people that care about the subject could start working on this.

**Reply by Shauren** [reply]
The amount of time we are willing to put into fixing meshes by hand is exactly 0

**Reply by ZaDarkSide** [reply]
> The amount of time we are willing to put into fixing meshes by hand is exactly 0

I'm glad you are speaking for the whole community in this matter. I hope everyone agrees with you. I for one I do think people can do what they want, I've seen even more tedious shit get done here by a couple of passionate dudes:  https://www.youtube.com/watch?v=fyJ2fl2IyxU

**Reply by Shauren** [reply]
I am speaking only for TrinityCore developers, not users

**Comment by Shauren** [comment]
> For all intents and purposes the map data is static and it's just gets added to it over time when new expansions come out

No, it is not. Old maps get modified all the time, for example in 9.1 human starting zone got modified

**Reply by ZaDarkSide** [reply]
Yeah, but i was mostly speaking about 3.3.5 and other already "finished" expansions/versions there are no new patches coming for 3.3.5 ever. master branch let's say it's the only exception, that's why the extractor will do most of the heavy lifting.

**Comment by jackpoz** [comment]
The main reason client related files (dbcs, maps, vmaps) are not available for download is to avoid as much as possible copyright issues as those files are extracted from the client. mmaps might be a grey area as they are generated, not extracted.

From a dev point of view, it costs little to generate these and upload them somewhere. From a user point of view, the time it takes to run the tools usually fit in 1 or 2 hours, depending on the CPU and disk performances. I never really understood what's the big deal with re-running tools.

Manual changes could be added to mmaps_generator, we already have some kind of changes: https://github.com/TrinityCore/TrinityCore/blob/587904ac68aae307a0d24ff77492a7ae1d4f7323/src/tools/mmaps_generator/MapBuilder.cpp#L1118-L1132

**Reply by ZaDarkSide** [reply]
Still having some static data already pre-generated and ready for download will always beat the ~2 hours of wait of generating that data.  The TDB releases are created for convenience also, no copyright issue there right?
And you are not asking people to run a program that can generate/sniff/extract needed DB data from retail, why isn't static map data treated the same way? It's seems a random option that TC has chosen to provide downloads for one thing and not the other.

**Reply by jackpoz** [reply]
Well, isn't that how it has always been ? With repacks being the ones providing dbcs/maps/vmaps and open source projects providing tools.
In a similar way we don't provide the client itself either.

**Reply by ZaDarkSide** [reply]
TDB isn't tools...and I didn't say anything about repacks, they suck in general.
Building code from source and downloading "official releases" from reputable open source groups like TC is your safest bet, but that's another discussion.

**Reply by mdX7** [reply]
Arctium (not TrinityCore) is now covering it here: https://tc.arctium.io/

**Reply by ZaDarkSide** [reply]
Amazing work guys as always, thx again Arctium!

**Comment by ZaDarkSide** [comment]
Okay I think I got enough feedback from you guys to understand there is not enough interest in the ideea I have proposed and the current way of always re-running the extractor tools is preferable. Thanks again for your comments.

**Reply by jackpoz** [reply]
Well we do have interested to making it easier to use TC, see for example https://ci.appveyor.com/project/DDuarte/trinitycore/branch/3.3.5/artifacts that provides pre-compiled binaries. We just don't feel like we can make everything available for legal reasons (whether it's just our fear or it's a shared one that's another discussion)

**Reply by ZaDarkSide** [reply]
> Well we do have interested to making it easier to use TC, see for example https://ci.appveyor.com/project/DDuarte/trinitycore/branch/3.3.5/artifacts that provides pre-compiled binaries. We just don't feel like we can make everything available for legal reasons (whether it's just our fear or it's a shared one that's another discussion)

That's neat, well take a look at Github Actions also, you are using Github after all, it's nice to have everything in one place.

**Comment by ZaDarkSide** [comment]
Seems somebody got the same idea to have some tooling around fixing the vmaps manually, you can check it out here: https://youtu.be/B5jbPtv_FKo?t=208
